src/glue.c

```diff
@@ -214,6 +217,135 @@ PyObject *mg_path_to_py_path(const mg_path *path) {
   return ret;
 }
 
+void maybe_decrement_ref(PyObject **obj) {
+  if (*obj) {
+    Py_DECREF(obj);
+  }
+}
+
+#define SCOPED_CLEANUP __attribute__((cleanup(maybe_decrement_ref)))
+#define IF_PTR_IS_NULL_PRINT_ERROR_AND_RETURN(ptr, value) \
+  if (!(ptr)) {                                           \
+    PyErr_Print();                                        \
+    return (value);                                       \
+  }
+
+PyObject *make_py_date(int y, int m, int d) {
+  PyDateTime_IMPORT;
+  PyObject *date = PyDate_FromDate(y, m, d);
+  if (!date) {
+    PyErr_Print();
+  }
+  return date;
+}
+
+PyObject *make_py_time(int h, int min, int sec, int mi) {
+  PyDateTime_IMPORT;
+  PyObject *time = PyTime_FromTime(h, min, sec, mi);
+  if (!time) {
+    PyErr_Print();
+  }
+  return time;
+}
+
+PyObject *make_py_datetime(int y, int m, int d, int h, int min, int sec,
+                           int mi) {
+  PyDateTime_IMPORT;
```

Comment:
This defines a static variable which is then used by each of datetime objects. Therefore, it either must be put in a global scope such that all instances have access to this function, or the way it is done now. I am unsure which one is the best.

Reply:
https://github.com/python/cpython/blob/main/Include/datetime.h#L200

Reply:
yes, the initialized variable is static so I presume that calling this function in `mgclientmodule.c` that is a different TU won't initialize the correct object for this TU. I would also like to avoid all the calls to import but I'm not sure how to achieve this.

Reply:
I don't know either, if we move it to `mgclientmodule.c` is segfaults (because of the way it's defined look at the above link)

Reply:
Ahh, I didn't know it is static. Then the current status is okay for me.